src/queryResult/utils.ts

```diff
@@ -86,16 +86,36 @@ export function registerCommonRequestHandlers(
         );
         if (
             currentState.isExecutionPlan &&
+            currentState.resultSetSummaries[message.batchId] &&
             // check if the current result set is the result set that contains the xml plan
             currentState.resultSetSummaries[message.batchId][message.resultId]
                 .columnInfo[0].columnName === Constants.showPlanXmlColumnName
         ) {
-            currentState.executionPlanState.xmlPlans =
-                // this gets the xml plan returned by the get execution
-                // plan query
-                currentState.executionPlanState.xmlPlans.concat(
-                    result.rows[0][0].displayValue,
-                );
+            currentState.executionPlanState.xmlPlans[
+                `${message.batchId},${message.resultId}`
+            ] = result.rows[0][0].displayValue;
+        }
+        // if we are on the last result set and still don't have any xml plans
+        // then we should not show the query plan. for example, this happens
+        // if user runs actual plan with all print statements
+        else if (
+            message.batchId ===
+                webviewViewController.recordLength(
+                    currentState.resultSetSummaries,
+                ) -
+                    1 &&
+            message.resultId ===
+                webviewViewController.recordLength(
+                    currentState.resultSetSummaries[message.batchId],
+                ) -
+                    1 &&
+            (!currentState.executionPlanState?.xmlPlans ||
+                !webviewViewController.recordLength(
+                    currentState.executionPlanState.xmlPlans,
+                ))
```

Comment:
Thanks for adding the comment block at the top!  For cascading boolean conditions like this one, I think it's _also_ nice to have comments on each "clause" in the conditional.

Might be worth checking to see if adding more parens tells Prettier to format it more nicely (it's weird that the `1` is on a different line from the measure it's being subtracted from)

Reply:
prettier removes "unnecessary" parens, so that's a no go

Reply:
for what it's worth, I do think moving the recordLength function to utils helps